src/fold/transformations/features.py

```diff
@@ -0,0 +1,114 @@
+from enum import Enum
+from typing import List, Union
+
+import holidays
+import pandas as pd
+
+from ..utils.list import wrap_in_list
+from .base import Transformation, fit_noop
+
+
+def get_weekends(dates: pd.DatetimeIndex) -> pd.Series:
+    return pd.Series((dates.weekday > 4).astype(int), index=dates)
+
+
+def encode_series(series: pd.Series) -> pd.Series:
+    return series.astype("category").cat.codes
+
+
+def get_multi_holidays(
+    dates: pd.DatetimeIndex, country_codes: List[str]
+) -> pd.DataFrame:
+    df = pd.DataFrame.from_dict(
+        {
+            country_code: holidays.country_holidays(
+                country_code,
+                years=dates.year.unique().to_list(),
+                language="en_US",
+            )
+            for country_code in country_codes
+        },
+    )
+
+    # Turn individual holiday names into category than reindex with the original dates
+    df = df.apply(encode_series).add(1).reindex(dates)
+
+    # When frequency is not daily we have to group and forward fill so that each hour/day/minute of the day has the same value
+    df["datetime"] = df.index
+    df = df.groupby(df["datetime"].dt.date).ffill().fillna(0.0)
```

Comment:
hmm, for me, the expected behaviour is that Christmas day would be present for all timestamps that have an overlap with Christmas day.
I think that's what would make sense for a model as well, imagine, a tabular model will need that info to differentiate between Christmas day midnight and Christmas day midnight + 1 minute. Otherwise, it'll take Christmas day + 1 minute as not Christmas day, although people's behaviour on Christmas day is different throughout the day, not just at midnight!

Reply:
That's what it is doing.

Reply:
If we don't do what's here, then this is what we get:
|     <index>      |    US    |
| ----- | ---  |
| 2018-12-25 00:00:00 | Christmas Eve |
| 2018-12-25 00:01:00 | NaN |
| 2018-12-25 00:02:00 | NaN  |
| 2018-12-25 00:03:00 | NaN  |
| 2018-12-25 00:04:00 | NaN  |
...

With grouping and `ffill()` we get:
|     <index>      |    US    |
| ----- | ---  |
| 2018-12-25 00:00:00 | Christmas Eve |
| 2018-12-25 00:01:00 | Christmas Eve |
| 2018-12-25 00:02:00 |  Christmas Eve   |
| 2018-12-25 00:03:00 |  Christmas Eve   |
| 2018-12-25 00:04:00 |  Christmas Eve   |
...

Reply:
I missed the important part in my explanation :)

Reply:
I didn't know you can do it this way, can you update the PR on how you would do it?

Reply:
I think I found a better method that also preserves the type of holiday.